collectors/feed/collectors.py

```diff
@@ -0,0 +1,281 @@
+import logging
+from concurrent.futures import ThreadPoolExecutor
+from io import BytesIO
+from itertools import chain
+
+import feedparser
+import requests
+import yaml
+from requests.exceptions import RequestException
+
+from osidb.models import Snippet
+
+from ..framework.models import KnownFeedEntry
+from .constants import (
+    CHROMIUM_REGEX,
+    FEEDS_FILE,
+    MAX_COLLECTOR_WORKERS,
+    REQUEST_TIMEOUT,
+)
+from .keywords import check_keywords
+from .models import FeedEntry, SkippedFeedEntry
+from .utils import find_cves, find_cvss, find_cwes, html_to_text
+
+log = logging.getLogger(__name__)
+
+# TODO: Refactor from the original source to a more efficient structure better matching OSIDB style
+#       - NOTE: If you diff this with the original, the changes are visible
```

Comment:
FYI, I'm working on refactoring it (what it inside a class, what is outside, what classes are there, etc). It won't be as directly comparable to the original code after the refactor.
For now, the value in non-refactored code is that it should be more obvious whether there are parts brought over that are either wrong or not needed anymore.